akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -155,6 +155,7 @@ private[akka] object FanIn {
       if (input.inputsDepleted) {
         if (marked(id)) markedDepleted += 1
         depleted(id) = true
+        onDepleted(id)
```

Comment:
The rationale was that it was depleted but the callback was never called. I considered this a bug.